examples/bond/test/bond-token-flow.js

```diff
@@ -0,0 +1,424 @@
+      globalInts: 8,
+      globalBytes: 15
+    };
+  });
+
+  const getGlobal = (key) => runtime.getGlobalState(applicationId, key);
+
+  // fetch latest account state
+  function syncAccounts () {
+    appManager = runtime.getAccount(appManager.address);
+    bondTokenCreator = runtime.getAccount(bondTokenCreator.address);
+    issuerAddress = runtime.getAccount(issuerAddress.address);
+    elon = runtime.getAccount(elon.address);
+    dex1 = runtime.getAccount(dex1.address);
+    dex2 = runtime.getAccount(dex2.address);
+    bob = runtime.getAccount(bob.address);
+  }
+
+  // Bond-Dapp initialization parameters
+  const appManagerPk = convert.addressToPk(appManager.address);
+  const issuePrice = 'int:1000';
+  const couponValue = 'int:20';
+  const maxIssuance = 'int:1000000';
+  const bondCreator = convert.addressToPk(bondTokenCreator.address);
+
+  it('Bond token application', () => {
+    /**
+    * Issue initial bond tokens to the issuer
+    * In epoch_0 elon buys 10 bonds
+    * In epoch 0 elon sells 2 bonds to bob for 2020 ALGO (in a group transaction)
+    * Manager creates dex 1
+    * Elon redeems his bonds (8), Elon buys 4 more bonds (so he will have 12 bonds in total)
+    * Manager creates dex 2
+    * Elon redeems all his bonds.
+    * Bob redeems his bonds from epoch 0 and 1
+    * Maturity period is set to 240 seconds(4 min) after the contract deployment.
+    * At maturity, manager creates and funds buyback and both elon and
+    * bob can exit all their tokens (12 and 2 respectively).
+    */
+
+    const currentBondIndex = runtime.addAsset(
+      'bond-token-0', { creator: { ...bondTokenCreator.account, name: 'bond-token-creator' } });
+
+    const creationFlags = Object.assign({}, flags);
+    const creationArgs = [
+      appManagerPk,
+      bondCreator,
+      issuePrice,
+      couponValue,
+      `int:${currentBondIndex}`,
+      maxIssuance
+    ];
+
+    // create application
+    applicationId = runtime.addApp(
+      { ...creationFlags, appArgs: creationArgs }, {}, approvalProgram, clearProgram);
+
+    // setup lsig account
+    // Initialize issuer lsig with bond-app ID
+    const scInitParam = {
+      TMPL_APPLICATION_ID: applicationId,
+      TMPL_OWNER: bondTokenCreator.address,
+      TMPL_APP_MANAGER: appManager.address
+    };
+    const issuerLsigProg = getProgram('issuer-lsig.py', scInitParam);
+    lsig = runtime.getLogicSig(issuerLsigProg, []);
+    issuerLsigAddress = lsig.address();
+
+    // sync escrow account
+    issuerAddress = runtime.getAccount(issuerLsigAddress);
+    console.log('Issuer Address: ', issuerLsigAddress);
+
+    // fund escrow with some minimum balance first
+    const fundEscrowParam = {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: master.account,
+      toAccountAddr: issuerLsigAddress,
+      amountMicroAlgos: minBalance + 10000,
+      payFlags: {}
+    };
+    runtime.executeTx(fundEscrowParam);
```

Comment:
Should we add a `fundLsig()` function in runtime for this? We use this other example tests as well. What do you think? We can do it in a new task.

Reply:
yes, we can. adding a task for it.

Reply:
https://www.pivotaltracker.com/story/show/179174388